changeset:   47875:d9d236a8d274
user:        Eli Friedman <sharparrow1@yahoo.com>
date:        Sun Jul 18 11:19:48 2010 +0200
summary:     Bug 579689 - Fix cases of return without expression in a function returning "int". r=khuey

diff -r d9555a5522cf -r d9d236a8d274 build/autoconf/moznbytetype.m4
--- build/autoconf/moznbytetype.m4	Sun Jul 18 11:17:14 2010 +0200
+++ build/autoconf/moznbytetype.m4	Sun Jul 18 11:19:48 2010 +0200
@@ -60,7 +60,7 @@
     AC_TRY_COMPILE([],
                    [
                      int a[sizeof ($type) == $2 ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_n_byte_type_$1=$type; break], [])
   done
@@ -91,7 +91,7 @@
     AC_TRY_COMPILE([],
                    [
                      int a[sizeof ($2) == $size ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_size_of_$1=$size; break], [])
   done
@@ -124,7 +124,7 @@
                    ],
                    [
                      int a[offsetof(struct aligner, a) == $align ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_align_of_$1=$align; break], [])
   done
diff -r d9555a5522cf -r d9d236a8d274 js/src/build/autoconf/moznbytetype.m4
--- js/src/build/autoconf/moznbytetype.m4	Sun Jul 18 11:17:14 2010 +0200
+++ js/src/build/autoconf/moznbytetype.m4	Sun Jul 18 11:19:48 2010 +0200
@@ -60,7 +60,7 @@
     AC_TRY_COMPILE([],
                    [
                      int a[sizeof ($type) == $2 ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_n_byte_type_$1=$type; break], [])
   done
@@ -91,7 +91,7 @@
     AC_TRY_COMPILE([],
                    [
                      int a[sizeof ($2) == $size ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_size_of_$1=$size; break], [])
   done
@@ -124,7 +124,7 @@
                    ],
                    [
                      int a[offsetof(struct aligner, a) == $align ? 1 : -1];
-                     return;
+                     return 0;
                    ],
                    [moz_cv_align_of_$1=$align; break], [])
   done

